cmd/ooniprobe/internal/ooni/ooni.go

```diff
@@ -180,7 +179,9 @@ func (p *Probe) Init(softwareName, softwareVersion, proxy string) error {
 
 	p.dbPath = utils.DBDir(p.home, "main")
 	log.Debugf("Connecting to database sqlite3://%s", p.dbPath)
-	db, err := database.Connect(p.dbPath)
+	db, err := database.NewDatabase(&database.DatabaseConfig{
+		DatabasePath: p.dbPath,
+	})
```

Comment:
I understand that your plan is that of adding to `DatabaseConfig` in the future a boolean field and later choose it to select which implementation to instantiate (the `ooniprobe` one or a `null` one). However, as things stand, I do not think we need to select which database to instantiate at runtime, rather we can decide that already at compile time: the ordinary database for `ooniprobe` and the null one for `miniooni`. As such, I do feel like a better pattern here would be the following:

```
func New(dbpath string) (*database.Database, error)
```

I am omitting `Database` here from `NewDatabase` because `database.New` is already readable while `database.NewDatabase` stutters a bit, as it repeats the word database.